source/speech/speech.py

```diff
@@ -3059,3 +3063,38 @@ def clearTypedWordBuffer() -> None:
 	complete the word (such as a focus change or choosing to move the caret).
 	"""
 	_curWordChars.clear()
+
+
+def _sapi4DeprecationWarning(synth: SynthDriver, audioOutputDevice: str, isFallback: bool):
+	"""A synthChanged event handler to alert the user about the deprecation of SAPI4."""
+
+	def setShown():
+		setattr(synth, "_hasWarningBeenShown", True)
```

Comment:
I think this should work
```suggestion
		synth._hasWarningBeenShown = true
```